core/src/main/java/io/kestra/core/services/FlowService.java

```diff
@@ -222,13 +222,15 @@ public static String cleanupSource(String source) {
         return source.replaceFirst("(?m)^revision: \\d+\n?","");
     }
 
-    public static String injectDisabledTrue(String source) {
-        Pattern p = Pattern.compile("^disabled\\s*:\\s*false\\s*", Pattern.MULTILINE);
+    public static String injectDisabledTrue(String source, Boolean disabled) {
+        String regex = disabled ? "^disabled\\s*:\\s*false\\s*" : "^disabled\\s*:\\s*true\\s*";
```

Comment:
```suggestion
        String regex = "^disabled\\s*:\\s*" + String.valueOf(!disabled) + "\\s*";
```

Reply:
Idk if that's really an issue but maybe we should remove the disabled property entirely instead of creating or updating it when we enable back the flow ?

Reply:
When we enable a flow, we filter by flows that are disabled so they already have the property (or the flow has been edited directly in the DB)